File: Frontend/src/app/models/Income/income.ts
import { Base } from "../Base.js";
import { Frequency } from "../Admin/frequency.js";
import { IncomeType } from "../Admin/income-type.js";
import { User } from "../Admin/user.js";
import { UserPaymentMethod } from "../Admin/UserPaymentMethod.js";

export class Income extends Base {

        
        public user? : User;
        public source? : IncomeType;
        public frequency? : Frequency;
        public received? : boolean;
        public value? : number;
        public receivedFrom? : string;
        public receivedIn? : UserPaymentMethod;
        public description?: string;
        public receivedOn?: Date;

}

File: Features/DL/Entities/Income.cs
namespace Features.DL.Entities
{
    public class Income: BaseModel
    {
        public virtual User User { get; set; }
        public virtual Frequency Frequency {  get; set; }

        public  virtual IncomeType Source { get; set; }

        public virtual Boolean Received { get; set; }

        public virtual Decimal Value { get; set; }

        public virtual string ReceivedFrom {  get; set; }

        public virtual DateTime? ReceivedOn { get; set; }

        public virtual UserPaymentMethod ReceivedIn {  get; set; }

        public virtual string Description { get; set; }
    }
}
